wordpress-mobile/WordPress-iOS
# Add Alternative App Icons

This PR adds some alternative app icons using iOS 10.3's `alternate icon` feature. This is just a fun feature to allow a bit of customisation for users. Here's the initial batch:

<img width="374" alt="Screenshot 2019-06-28 at 11 33 51" src="https://user-images.githubusercontent.com/4780/60336621-c7f03e00-9998-11e9-8ff8-ab4e33a19327.png">

![icon-change-1](https://user-images.githubusercontent.com/4780/60298412-5f667a00-9922-11e9-825e-065e800ec006.gif)

The app's icon can be changed by going to **Me** > **App Settings** > **App Icon** (down the bottom), and choosing an icon. Some notes on implementation:

* Alternate icons have to be listed in the Info.plist. Unfortunately, they can't be included in an asset catalog, so they have to be added to the project the old fashioned way as standard files.
* I added code in `AppIconViewController.swift` to dynamically fetch the list of icon names from the Info.plist instead of hardcoding them directly.
* I added some tracking as it'd be nice to see how much use this feature gets, but I didn't want to link specific users to specific icons so I'm just sending generic 'icon changed' and 'icon reset' events (associated PR here: https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/211).
* The Jetpack icon is only visible in the picker if you have a Jetpack site added to the app.

**To test:**

* Build and run
* Navigate to Me > App Settings > App Icon, and try changing icons
* Ensure you test switching back to the original icon as that's a slightly different case
* Check the Jetpack icon appears if you have a Jetpack site, and doesn't appear for a normal self-hosted site (you can create one with https://poopy.life if you don't have one).

Update release notes:
